bug_update.php

```diff
@@ -412,11 +412,25 @@
 	custom_field_set_value( $t_custom_field_to_set['id'], $f_bug_id, $t_custom_field_to_set['value'] );
 }
 
-# Add a bug note if there is one.
-if( $t_bug_note->note || helper_duration_to_minutes( $t_bug_note->time_tracking ) > 0 ) {
-	$t_bugnote_id = bugnote_add( $f_bug_id, $t_bug_note->note, $t_bug_note->time_tracking, $t_bug_note->view_state == VS_PRIVATE, 0, '', null, false );
-	bugnote_process_mentions( $f_bug_id, $t_bugnote_id, $t_bug_note->note );
-}
+# Add a bug note and attachments if there are any.
+$f_files = gpc_get_file( 'ufile', array() );
+$t_note_query = array( 'issue_id' => $f_bug_id );
+$t_note_payload = array(
+	'text' => $t_bug_note->note,
+	'view_state' => array(
+		'id' => $t_bug_note->view_state
+	),
+	'time_tracking' => array(
+		'duration' => $t_bug_note->time_tracking
+	),
+	'files' => helper_array_transpose( $f_files )
+);
+$t_note_data = array(
+	'query' => $t_note_query,
+	'payload' => $t_note_payload,
+);
+$t_note_command = new IssueNoteAddCommand( $t_note_data );
```

Comment:
The commands should reflect the intention of the user and not be used as re-usable code.  For example, the user intent it to change status or update issue in this case.  Using command here may trigger emails, events, and other semantics that is not expected.  The #1400 PR was attempting at having an update command.

Anyway, whether we have an update command or not, the re-usable code between add note and update issue should be in APIs that are called by both commands.

Reply:
If not using the command, we are at the previous situation: file api don't provide a way to add an attachment that satisfies the expected internal representation.